### PR TITLE
dnssec daemons: read the dns context config file for debug state

### PR DESCRIPTION
**Pull request by rcritten**
This had been hardcoded to debug=True but it spams the logs
with a lot of unnecessary information.

Allow it to be enabled for troubleshooting purposes but keep it
disabled by default.

Enabling debug would involve created /etc/ipa/dns.conf:

[global]
debug = True

I didn't add a more generic mechanism because for now we only need
the value of debug and it introduces a lot of type conversion
headaches. ipalib handles this automatically but to duplicate this
would be corner-case city.

Fixes: https://pagure.io/freeipa/issue/9128

Signed-off-by: Rob Crittenden <rcritten@redhat.com>

### CODE REVIEW

**Comment by rcritten** [comment]
Note that I'm a bit torn on what to test here. I can write a unit test for the new function pretty easily though I'm not sure this is something likely to regress as its only used in 3 places.
For integration testing it seems fragile to ensure there are no DEBUG lines in the journal for the service.

**Comment by flo-renaud** [comment]
Hi @rcritten 
if I understand correctly, the new function `get_config_debug()` is needed because we are calling standard_logging_setup before the call to bootstrap the API. If we had something in the reverse order:
```
ipalib.api.bootstrap(context='dns', confdir=paths.ETC_IPA, in_server=True)
standard_logging_setup(debug=ipalib.api.env.debug)
```
it would allow to set the level for the calls happening after those lines but we would be missing the debug info corresponding to the bootstrap.

My question: do we often have issues during the bootstrap, that would justify the debug level? Or is the debug level useful mainly for what happens after bootstrap in the daemon?

**Comment by rcritten** [comment]
@flo-renaud yes, this exactly. It's a chicken-and-egg problem.
I erred on the side of caution in duplicating a piece of the configuration reading code (ipalib cannot be imported). I can't predict how this could fail on startup. I imagine that it would be fine enabling logging after-the-fact but enabling debugging during API init would provide more context, I think.
The main downside beyond another dozen lines to support forverer is it doesn't do the inheritance that a context usually provides (e.g. read default.conf, then context.conf). I think this is a reasonable trade-off.
The elephant in the room is how to discover that you can even create /etc/ipa/dns.conf with the right contents to enable debugging.

**Comment by flo-renaud** [comment]
@rcritten the code looks good to me, it properly removes the debug level by default.
We have some information re. debug level in default.conf man page, maybe dns.conf can also be documented there, in the last section **FILES?**

**Comment by rcritten** [comment]
@flo-renaud maybe? It is a bit hand-wavy now and not all contexts are documented.
My biggest concern is this: someone is having a problem with dnssec so they want to increase debugging. These daemons ship in libexec so have no man pages. So how does someone go from that to default.conf(5), or anywhere really, to find dns.conf? With the correct formatting.
I suppose I could add a line at startup that says something like "to increase debugging see default.conf(5)"?

**Comment by rcritten** [comment]
> That's a good idea. The page https://www.freeipa.org/page/Troubleshooting/DNS can also be updated with this information, and we could also add a troubleshooting section to https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/working_with_dns_in_identity_management/index

Patch updated. Once accepted I can update the upstream docs and see about updating the official docs. Timing will be important.

File: daemons/dnssec/ipa-dnskeysync-replica.in [hunk 1]
```diff
@@ -145,7 +146,11 @@ def ldap2replica_zone_keys_sync(ldapkeydb, localhsm):
 
 
 # IPA framework initialization
-standard_logging_setup(debug=True)
+debug = get_config_debug('dns')
+standard_logging_setup(debug=debug)
+if not debug:
```
Comment (flo-renaud) [inline daemons/dnssec/ipa-dnskeysync-replica.in:151]: By default, only errors messages are logged to the console: https://github.com/freeipa/freeipa/blob/ce0592bd478c184b1c0c2a75f675c5a361f4a9a4/ipapython/ipa_log_manager.py#L92-L99
To have logger.info write in the console, `standard_logging_setup` has to be called with `(debug=debug, verbose=True)`.

Comment (abbra) [reply daemons/dnssec/ipa-dnskeysync-replica.in:151]: @rcritten could you please update this too?

Comment (rcritten) [reply daemons/dnssec/ipa-dnskeysync-replica.in:151]: Sorry, I missed this comment. The debug level is higher than info so all message should be logged. I don't believe we need to set verbose as debug does the same thing.

Comment (flo-renaud) [reply daemons/dnssec/ipa-dnskeysync-replica.in:151]: The intent of line 152 was to print the guidance in all cases (even if debug is not set), but it's using `logger.info`. If debug is not set in the `dns.conf` file, then the console prints only ERROR messages and the INFO message `To increase debugging...` never goes to the journal.

**Comment by flo-renaud** [comment]
@rcritten 
thanks for the patch, LGTM. We can now see the line `To increase debugging set debug=True in dns.conf...` in the journal even without setting debug level. ACK.

**Comment by flo-renaud** [comment]
master:

* cef6a90288d9ec0c69f74ddce93db14694ba7a76 dnssec daemons: read the dns context config file for debug state
